context.go

```diff
@@ -24,6 +28,14 @@ type Context struct {
 // NewContext creates a new context. For use in when invoking an App or Command action.
 func NewContext(app *App, set *flag.FlagSet, parentCtx *Context) *Context {
 	c := &Context{App: app, flagSet: set, parentContext: parentCtx}
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		defer cancel()
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+		<-sigs
```

Comment:
I don't think a random library should yank signal handling of the whole process because it feels like it.
It should be up to the user to set this up.

Reply:
@AudriusButkevicius similar to the comment above, this follows the net/http request Context pattern. If the user wants to use their own context, they can simple create one and ignore the given one, or they can wrap the given one. 

Furthermore, the Context is exposed, so the user can replace it. 

However, if you'd like me to change that to just be context.Background() and have each user implement the boiler plate above on their side, that's okay with me too :+1:

Reply:
This was not about that, but about hijacking the processes signal handlers, and net/http not doing that by default because it's being used.

It seems go abstracts signal handling, so calling notify multiple times does not override the process signal handler repeatedly, in which case I am ok this this.